lib/diplomat/check.rb

```diff
@@ -11,7 +11,27 @@ def checks
       JSON.parse(ret.body)
     end
 
-    # Register a check
+    # Register a HTTP check
+    # @param check_id [String] the unique id of the check
+    # @param name [String] the name
+    # @param notes [String] notes about the check
+    # @param script [String] command to be run for check
+    # @param interval [String] frequency (with units) of the check execution
+    # @param ttl [String] time (with units) to mark a check down
+    # @return [Integer] Status code
+    #
+    def register_http(check_id, name, notes, url, method, header, interval)
```

Comment:
There are lots of parameters to this function, would you mind using named parameters instead ?

I would rather use this signature:
```ruby
def register_http(url, interval, id: nil, id:name, notes: nil, method: 'GET', header: nil)
  [...]
end
```
since both interval and url are mandatory (see https://www.consul.io/api/agent/check.html#register-check )
It would allow to use only the provided parameters.

It also lacks `header` in ruby documentation as well as calling it `headers` would probably be more accurate (and specify type)